pkg/i2gw/ingress2gateway.go

```diff
@@ -78,13 +84,40 @@ func ToGatewayAPIResources(ctx context.Context, namespace string, inputFile stri
 	for _, provider := range providerByName {
 		ir, conversionErrs := provider.ToIR()
 		errs = append(errs, conversionErrs...)
+
+		// Run implementations on the IR before rendering GatewayResources.
+		var implObjs []client.Object
+		for _, implName := range implementations {
+			emitter, ok := ImplementationEmitters[implName]
+			if !ok {
+				continue // already validated in CLI
+			}
+			objs, err := emitter.Emit(&ir)
+			if err != nil {
+				errs = append(errs, field.InternalError(field.NewPath("implementation", implName), err))
+			}
+			implObjs = append(implObjs, objs...)
+		}
+
 		providerGatewayResources, conversionErrs := provider.ToGatewayResources(ir)
 		errs = append(errs, conversionErrs...)
+
+		// Convert client.Objects to unstructured and append to GatewayExtensions.
+		for _, obj := range implObjs {
```

Comment:
Ok, I was thinking about this but want to get your throughts.

Right now, GCE is the only provider that takes advantage of provider-specific extensions. Do you think that we could have a GCE case and a non-GCE case?